rust/private/rust_analyzer.bzl

```diff
@@ -23,22 +23,26 @@ to Cargo.toml files.
 load("//rust/platform:triple_mappings.bzl", "system_to_dylib_ext", "triple_to_system")
 load("//rust/private:common.bzl", "rust_common")
 load("//rust/private:rustc.bzl", "BuildInfo")
-load("//rust/private:utils.bzl", "find_toolchain")
+load("//rust/private:utils.bzl", "dedent", "find_toolchain")
 
 RustAnalyzerInfo = provider(
     doc = "RustAnalyzerInfo holds rust crate metadata for targets",
     fields = {
         "build_info": "BuildInfo: build info for this crate if present",
         "cfgs": "List[String]: features or other compilation --cfg settings",
+        "crate_specs": "List[File]: transitive closure of OutputGroupInfo files",
         "crate": "rust_common.crate_info",
         "deps": "List[RustAnalyzerInfo]: direct dependencies",
         "env": "Dict{String: String}: Environment variables, used for the `env!` macro",
         "proc_macro_dylib_path": "File: compiled shared library output of proc-macro rule",
-        "transitive_deps": "List[RustAnalyzerInfo]: transitive closure of dependencies",
     },
 )
 
 def _rust_analyzer_aspect_impl(target, ctx):
+    if OutputGroupInfo in target:
```

Comment:
Why do we have to "re-provide" output groups here?

Reply:
This is a little bit of a hack to ensure that the rust_analyzer aspect generates the rust_analyzer_sysroot_src, which doesn't have crate_info so will be skipped by the check on L46. We could use a separate aspect for it though.

More generally, we need to re-provide the output groups because otherwise if you did something like `bazel build --aspects=@rules_rust//rust:defs.bzl%rust_analyzer_aspect --output_groups=rust_analyzer_crate_spec //path/to/single:target` then only the crate_spec file for `//path/to/single:target` will be generated, but none of its deps. Re-providing OutputGroupInfo ensures that all the dependent crate_spec files are generated as well.